scvi/hub/hub_model.py

```diff
@@ -0,0 +1,244 @@
+import importlib
+import logging
+import os
+from pathlib import Path
+from typing import List, Optional, Type, Union
+
+import anndata
+import rich
+import torch
+from anndata import AnnData
+from huggingface_hub import (
+    HfApi,
+    ModelCard,
+    ModelFilter,
+    create_repo,
+    snapshot_download,
+)
+from huggingface_hub.hf_api import ModelInfo
+from rich.markdown import Markdown
+
+from scvi.data._download import _download
+from scvi.hub.hub_metadata import HubMetadata, HubModelCardHelper
+from scvi.model.base import BaseModelClass
+
+from ._constants import (
+    HF_LIBRARY_NAME,
+    MAX_HF_UPLOAD_SIZE,
+    METADATA_FILE_NAME,
+    MODEL_CARD_FILE_NAME,
+)
+
+logger = logging.getLogger(__name__)
+
+
+class HubModel:
+    """Placeholder docstring. TODO complete."""
+
+    def __init__(
+        self,
+        local_dir: str,
+        metadata: Optional[Union[HubMetadata, str]] = None,
+        model_card: Optional[Union[HubModelCardHelper, ModelCard, str]] = None,
+    ):
+        self._local_dir = local_dir
+
+        self._model_path = f"{self._local_dir}/model.pt"
+        self._adata_path = f"{self._local_dir}/adata.h5ad"
```

Comment:
do we already have a util to load these?

Reply:
not that I know of. Please LMK if there is one